dev/dll/XamlControlsResources.cpp

```diff
@@ -118,11 +118,19 @@ void XamlControlsResources::UpdateSource()
         }()
     };
 
-    // Because of Compact, UpdateSource may be executed twice, but there is a bug in XAML and manually clear theme dictionaries here:
-    // Prior to RS5, when ResourceDictionary.Source property is changed, XAML forgot to clear ThemeDictionaries.
-    ThemeDictionaries().Clear();
+    // Because of Compact, UpdateSource may be executed twice, so we need to add this as a merged dictionary -
+    // calling Application.LoadComponent on the same element will throw an exception.
+    winrt::ResourceDictionary dictionary;
+    winrt::Application::LoadComponent(dictionary, uri, winrt::ComponentResourceLocation::Nested);
 
-    Source(uri);
+    uint32_t existingDictionaryIndex = 0;
+    if (m_xamlControlsResourcesDictionary && MergedDictionaries().IndexOf(m_xamlControlsResourcesDictionary.get(), existingDictionaryIndex))
+    {
+        MergedDictionaries().RemoveAt(existingDictionaryIndex);
+    }
+
+    m_xamlControlsResourcesDictionary.set(dictionary);
+    MergedDictionaries().InsertAt(0, m_xamlControlsResourcesDictionary.get());
```

Comment:
Right, I should have talked about this in the PR description.  Setting the Source like we were doing makes it so x:Uia doesn't work properly.  There are two different places that we can look for localized strings - either from the application resources, or from component resources.  The default is application, and the only way to override that is via Application.LoadComponent.  Using Application.LoadComponent instead of setting the Source property gives us the right information to know where to load the strings from to properly support x:Uia.